h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -594,12 +589,12 @@ private void pollAndUpdateProgress(Stage stage, String name, WorkAllocations.Wor
       }
 
       if (JobType.HyperparamSearch == work.type) {
-        Grid grid = (Grid)subJob._result.get();
+        Grid<?> grid = (Grid)subJob._result.get();
         int gridCount = grid.getModelCount();
-        if (gridCount > gridLastCount) {
-          userFeedback.info(Stage.ModelTraining, "Built: " + gridCount + " models for search: " + name);
+        if (gridCount > lastGridCount) {
```

Comment:
@seb-h2o couple of questions for pollAndUpdateProgress method: 
1) Why we are checking for `HyperparamSearch` and `ModelBuild` work types for stragglers and only for `HyperparamSearch` in the `while (subJob.isRunning())` block?
2) Following piece of code is almost identical
```
Grid<?> grid = (Grid) subJob._result.get();
        int gridCount = grid.getModelCount();
        if (gridCount > lastGridCount) {
          userFeedback.info(stage, "Built: " + gridCount + " models for search: " + name);
          this.addModels(grid.getModelKeys());
          lastGridCount = gridCount;
        }
```
why we call `subJob._result.get()` instead of just `subJob.get()` ?  Aren't we checking the same case twice here? If it is for the same case but we need to check it at different moments then we need to move it to a method and call it twice. 
3) if `gridCount` is total number of grid entries then `gridCount > lastGridCount` will always be true except for the case when they are equal. And why don't we want to add model and feedback for last grid count?
4) if I got naming correctly for `lastGridCount` then maybe we can use `lastGridEntry'? wdyt?

Reply:
@deil87 
1. from my understanding, `ModelBuild` build only one model, so we don't need to pick it live, whereas for grids it makes sense (live progress is only available in Flow today, but this may/will change). Collecting stragglers also makes sense for grids due to the 1s sleep: after this sleep the last model may be completed and the job stopped, which will escape the while loop, so you need to check for this last ones....
2. `subJob.get()` blocks until completion, whereas `subJob._result.get()` is a way to peep into the result, which makes sense inside the loop as we want to see live updates.
3. correct for the first part. The "feedback" can surely be improved here and in various places, what do you mean by "add model and feedback for last grid count"? Currently, it's adding model and feedback each time there's one or more new models since last update, if you were testing for '=', then the "feedback" would be printed after every 1s sleep...
4. not sure I get your point here, entry would mean it stores a model, I think count makes sense here.

Reply:
@seb-h2o 3) I see now. Comment for `grid.getModelCount();` is confusing. I thought it is a total number of entries in the grid, but it looks like it is number of (evaluated+launched) entries so far. 
4) Then of course `lastGridEntry` is not appropriate here. But still I think `lastGridCount` could be improved. It could be read as `count from last grid in the sequence of grids`. I would suggest something that convey information that it is about models that have been built...even method that you are calling says `grid.getModelCount()` so it is more about modelsCount than gridCount... for example for gridCount we can use `evaluatedModelsCount' or  `countOfBuiltModelsInGrid` or `evaluatedModelsCountInGrid`. For `lastGridEntry` - same but with `last` prefix.   Longer but more clear.

Reply:
you're getting picky here, it's a local variable whose scope is limited to the loop ;)
but yeah, `lastGridModelsCount` is better... I was just reflecting the existing local variable named `gridCount`...

Reply:
@seb-h2o regarding limited scope. If this was in the dedicated method with the naming that reflects what we are doing here...then yes. But this method is 82 lines long and name of the method does not say anything about entities we are polling and tracking. Sorry for being picky... but it is for good :)

Reply:
it's also legacy code: anyway, now that I need to make another push, will rename this